web_programming/crawl_google_results.py

```diff
@@ -1,24 +1,32 @@
-import sys
 import webbrowser
+from sys import argv
+from urllib.parse import quote
 
-import requests
 from bs4 import BeautifulSoup
-from fake_useragent import UserAgent
+from requests import get
 
 if __name__ == "__main__":
+    if len(argv) > 1:
+        query = "%20".join(argv[1:])
+    else:
+        query = quote(str(input("Search: ")))
+
     print("Googling.....")
-    url = "https://www.google.com/search?q=" + " ".join(sys.argv[1:])
-    res = requests.get(url, headers={"UserAgent": UserAgent().random})
-    # res.raise_for_status()
-    with open("project1a.html", "wb") as out_file:  # only for knowing the class
-        for data in res.iter_content(10000):
-            out_file.write(data)
-    soup = BeautifulSoup(res.text, "html.parser")
-    links = list(soup.select(".eZt8xd"))[:5]
 
-    print(len(links))
-    for link in links:
-        if link.text == "Maps":
-            webbrowser.open(link.get("href"))
-        else:
-            webbrowser.open(f"http://google.com{link.get('href')}")
+    url = f"https://www.google.com/search?q={query}&num=2"
+
+    res = get(
+        url,
+        headers={
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0"
```

Comment:
Tests are failing on web_programming/crawl_google_results.py:21:89: E501 line too long (106 > 88 characters)

Why is this version an improvement on the original?

Reply:
> Why is this version an improvement on the original?

Yeah!

Reply:
Why is it better?

Reply:
> Why is it better?

### The previous or original one Simply type your query into Google. But this one actually scrapes the result link from Google and opens it for you. As a result, you will save time.

### Running the previous or original program
```bash
C:\> python crawl_google_results.py how to code on ipad
Googling.....
3
```
It opens these three links, or urls, shown below:
- https://www.google.com/search?q=how+to+code+on+ipad&ie=UTF-8
- https://www.google.com/search?q=how+to+code+on+ipad&ie=UTF-8&tbm=isch
- https://www.google.com/search?q=how+to+code+on+ipad&ie=UTF-8&tbm=shop

---

### Running my program
```bash
C:\> python crawl_google_results.py how to code on ipad
Googling.....
```
It opens the search result link or url.
- https://coder.com/blog/a-guide-to-writing-code-on-an-ipad

Reply:
Understood.  Let's move these changes to a new, separate file like `launch_google_results.py` or `open_google_results.py`.

Reply:
Yeah! sounds good

Reply:
Created [`web_programming/open_google_results.py `](https://github.com/TheAlgorithms/Python/pull/7085/files#diff-f4aa5fc454a60b2207bbb9c15442628be868de11b665d2874b3f7781abbb75bb)

Reply:
@cclauss So can you merge now?